bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -5511,14 +5528,17 @@ void ReplicaImp::executeRequestsInPrePrepareMsg(concordUtils::SpanWrapper &paren
     auto epoch = bftEngine::EpochManager::instance().getSelfEpochNumber();
     bftEngine::EpochManager::instance().setSelfEpochNumber(epoch);
     bftEngine::EpochManager::instance().setGlobalEpochNumber(epoch);
-    Digest stateDigest;
-    Digest otherDigest;
+    Digest stateDigest, reservedPagesDigest, rvbDataDigest;
     CheckpointMsg::State state;
     const uint64_t checkpointNum = lastExecutedSeqNum / checkpointWindowSize;
-    stateTransfer->getDigestOfCheckpoint(
-        checkpointNum, sizeof(Digest), state, (char *)&stateDigest, (char *)&otherDigest);
-    CheckpointMsg *checkMsg =
-        new CheckpointMsg(config_.getreplicaId(), lastExecutedSeqNum, state, stateDigest, otherDigest, false);
+    stateTransfer->getDigestOfCheckpoint(checkpointNum,
+                                         sizeof(Digest),
+                                         state,
+                                         stateDigest.content(),
+                                         reservedPagesDigest.content(),
+                                         rvbDataDigest.content());
+    CheckpointMsg *checkMsg = new CheckpointMsg(
```

Comment:
Same here, use of std::shared_ptr is recommended.

Reply:
See my comment above.